File: RMS_Flutter/lib/model/Bill.dart


import 'package:rms_flutter/model/user.dart';

class BillModel {
  int? id;
  int? totalAmount;
  String? billDate;
  Null? paymentMethod;
  String? status;
  User? paidBy;
  User? receivedBy;

  BillModel(
      {this.id,
        this.totalAmount,
        this.billDate,
        this.paymentMethod,
        this.status,
        this.paidBy,
        this.receivedBy});

  BillModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    totalAmount = json['totalAmount'];
    billDate = json['billDate'];
    paymentMethod = json['paymentMethod'];
    status = json['status'];
    paidBy =
    json['paidBy'] != null ? new User.fromJson(json['paidBy']) : null;
    receivedBy = json['receivedBy'] != null
        ? new User.fromJson(json['receivedBy'])
        : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['totalAmount'] = this.totalAmount;
    data['billDate'] = this.billDate;
    data['paymentMethod'] = this.paymentMethod;
    data['status'] = this.status;
    if (this.paidBy != null) {
      data['paidBy'] = this.paidBy!.toJson();
    }
    if (this.receivedBy != null) {
      data['receivedBy'] = this.receivedBy!.toJson();
    }
    return data;
  }
}

File: RMS_Spring/src/main/java/com/Project/RMSSpring/entity/Bill.java
package com.Project.RMSSpring.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.*;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "bills")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Bill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private double totalAmount;

    @Column(nullable = false)
    private LocalDateTime billDate;

    @Column(length = 20)
    private String paymentMethod; // Cash, Card

    @Column(nullable = false)
    private String status;  // PAID, UNPAID, CANCELLED

    // User Relationship (Bill belongs to a user who places an order)
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User paidBy;

    // Admin Relationship (Admin who approves the bill)
    @ManyToOne
    @JoinColumn(name = "admin_id")
    private User receivedBy;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @OneToOne
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;
}
